test/zero-downtime/sequences_test.go

```diff
@@ -0,0 +1,205 @@
+				suite.Run(t, s)
+				wgSequences.Done()
+			}()
+
+		}
+	}
+	wgSequences.Wait()
+	env.Wg.Done()
+}
+
+func (suite *TestSuiteSequences) Test_Evaluation() {
+	var finished *models.KeptnContextExtendedCE
+	suite.env.failSequence()
+	suite.T().Log("deleting lighthouse configmap from previous test run")
+	_, _ = testutils.ExecuteCommand(fmt.Sprintf("kubectl delete configmap -n %s lighthouse-config-%s", testutils.GetKeptnNameSpaceFromEnv(), suite.project))
+
+	//// now let's add an SLI provider
+	suite.T().Log("adding SLI provider")
+	_, err := testutils.ExecuteCommand(fmt.Sprintf("kubectl create configmap -n %s lighthouse-config-%s --from-literal=sli-provider=my-sli-provider", testutils.GetKeptnNameSpaceFromEnv(), suite.project))
+	suite.Nil(err)
+	_, finished = testutils.PerformResourceServiceTest(suite.T(), suite.project, "myservice", true)
+	if finished.Data != nil {
+		suite.env.passFailedSequence()
+	}
+}
+
+func (suite *TestSuiteSequences) Test_DeliveryFails() {
+	suite.trigger("delivery", nil, false)
+}
+
+func (suite *TestSuiteSequences) Test_ExistingEvaluationFails() {
+	suite.trigger("evaluation", nil, true)
+}
+
+func (suite *TestSuiteSequences) trigger(triggerType string, data keptn.EventProperties, existing bool) {
+	project := suite.project
+	if existing {
+		project = suite.env.ExistingProject
+	}
+
+	suite.T().Logf("triggering sequence %s for project %s", triggerType, project)
+	// trigger a delivery sequence
+	keptnContext, err := testutils.TriggerSequence(project, "myservice", "hardening", triggerType, data)
+	suite.Nil(err)
+	suite.T().Logf("triggered sequence %s for project %s with context %s", triggerType, project, keptnContext)
+	sequence := NewTriggeredSequence(keptnContext, project, triggerType)
+
+	suite.checkSequence(sequence)
+}
+
+func (suite *TestSuiteSequences) checkSequence(sequence TriggeredSequence) {
+
+	var sequenceFinishedEvent *models.KeptnContextExtendedCE
+	stageSequenceName := fmt.Sprintf("%s.%s", "hardening", sequence.sequenceName)
+	var err error
+
+	suite.T().Logf("verifying completion of sequence %s with keptnContext %s in project %s", sequence.sequenceName, sequence.keptnContext, sequence.projectName)
+	suite.Eventually(func() bool {
+		sequenceFinishedEvent, err = testutils.GetLatestEventOfType(sequence.keptnContext, sequence.projectName, "hardening", v0_2_0.GetFinishedEventType(stageSequenceName))
+		if sequenceFinishedEvent == nil || err != nil {
+			return false
+		}
+		suite.env.passSequence()
+		return true
+	}, 1*time.Minute, 5*time.Second)
+
+	if sequenceFinishedEvent == nil || err != nil {
+		suite.env.failSequence()
+		suite.T().Errorf("sequence %s with keptnContext %s in project %s has NOT been finished", sequence.sequenceName, sequence.keptnContext, sequence.projectName)
+
+	} else {
+		suite.T().Logf("sequence %s with keptnContext %s in project %s has been finished", sequence.sequenceName, sequence.keptnContext, sequence.projectName)
+	}
+}
+
+func GetShipyard() (string, error) {
```

Comment:
Is there a reason why we don't add a simple shipyard file string instead of creating the structure, converting it to a string and creating tmp file afterwards ? Seems like it is a very simple one, so it shouldn't be a problem...

Reply:
also I would expect this method to be present in `test/zero-downtime/testsuite_zerodowntime_test.go`

Reply:
sure it can be done like that